x-pack/legacy/plugins/siem/server/lib/detection_engine/signals_mapping.json

```diff
@@ -82,6 +82,9 @@
               "tags": {
                 "type": "keyword"
               },
+              "threats": {
+                "type": "object"
+              },
```

Comment:
This part here should be directly from ECS rather than object so we get KQL support.

So you want this to be:

```json
      "threat": {
        "properties": {
          "framework": {
            "type": "keyword"
          },
          "tactic": {
            "properties": {
              "id": {
                "type": "keyword"
              },
              "name": {
                "type": "keyword"
              },
              "reference": {
                "type": "keyword"
              }
            }
          },
          "technique": {
            "properties": {
              "id": {
                "type": "keyword"
              },
              "name": {
                "type": "keyword"
              },
              "reference": {
                "type": "keyword"
              }
            }
          }
        }
      },
```

Before shipping we will probably change this file to be two different parts. One JSON which is directly from the ECS generated mapping as-is and then a TypeScript file that imports it and adds the extra signal mapping in as that will be more maintainable and upgrades should be simpler.

Reply:
Without nesting this seems to work ok, so I am fine with it where it is at unless we see other issues.